documentation/src/main/asciidoc/topics/proc_implementing_serial_cxt_initializer.adoc

```diff
@@ -0,0 +1,38 @@
+[id='implementing_proto_marshallers']
+= Manually Implementing Serialization Context Initializers
+In some cases you might need to manually define `.proto` schema files and
+implement ProtoStream marshallers. For example, if you cannot modify Java
+object classes to add annotations.
+
+.Procedure
+
+. Create a `.proto` schema with Protobuf messages.
++
+[source,proto]
+----
+include::config_examples/library.proto[]
+----
++
+. Use the `org.infinispan.protostream.MessageMarshaller` interface to implement
+marshallers for your classes.
++
+[source,java]
+.BookMarshaller.java
+----
+include::code_examples/BookMarshaller.java[]
+----
++
+[source,java]
+.AuthorMarshaller.java
+----
+include::code_examples/AuthorMarshaller.java[]
+----
++
+. Create a `SerializationContextInitializer` implementation that registers the
+`.proto` schema and marshallers with a `SerializationContext`.
```

Comment:
Maybe make it clear that we're talking about ProtoStream Marshallers here. Something like:

"schema and the ProtoStream marshaller implementations".